modules/AlphaWalletFoundation/AlphaWalletFoundation/Transactions/TransactionsService.swift

```diff
@@ -142,17 +130,31 @@ public class TransactionsService {
         }
     }
 
-    @objc private func stopTimers() {
+    @objc private func stop() {
         for each in providers {
-            each.value.stopTimers()
+            each.value.stop()
         }
     }
 
-    @objc private func restartTimers() {
+    @objc private func resume() {
         guard !config.development.isAutoFetchingDisabled else { return }
 
         for each in providers {
-            each.value.runScheduledTimers()
+            each.value.resume()
+        }
+    }
+    //TODO: call when receive a push notification
+    public func forceResumeOrStart(server: RPCServer) {
+        guard let provider = providers[server] else { return }
+
+        switch provider.state {
```

Comment:
I think I understand, but the functions called in the `switch` statement reads a little strange, eg. `stop()` followed by `resume()` when it's `.running`. Usually it's `stop()` and then `start()`. Similarly calling `resume()` when it's `.stopped`.

Or how about changing `func stop()` to `func pause()` if it makes sense? Reads a little better?

Reply:
> Or how about changing func stop() to func pause() if it makes sense? Reads a little better?

sure